project/cmake/scripts/common/addon-helpers.cmake

```diff
@@ -137,6 +140,20 @@ macro (build_addon target prefix libs)
     INSTALL(TARGETS ${target} DESTINATION ${CMAKE_INSTALL_LIBDIR}/addons/${target})
     INSTALL(DIRECTORY ${target} DESTINATION share/${APP_NAME_LC}/addons PATTERN "addon.xml.in" EXCLUDE)
   ENDIF(PACKAGE_ZIP OR PACKAGE_TGZ)
+  if(KODI_BUILD_DIR)
+    file(GLOB_RECURSE files ${CMAKE_CURRENT_SOURCE_DIR}/${target}/*)
+    foreach(file ${files})
+      string(REPLACE "${CMAKE_CURRENT_SOURCE_DIR}/${target}/" "" name "${file}")
+      # A good way to deal with () in filenames
+      if(NOT ${file} MATCHES addon.xml.in)
+        configure_file(${file} ${KODI_BUILD_DIR}/addons/${target}/${name} COPYONLY)
+      endif()
+    endforeach()
+    add_custom_command(TARGET ${target} POST_BUILD
+                       COMMAND ${CMAKE_COMMAND} -E copy
+                                $<TARGET_LINKER_FILE:${target}>
+                                ${KODI_BUILD_DIR}/addons/${target}/$<TARGET_LINKER_FILE_NAME:${target}>)
```

Comment:
@notspiff: I'm wondering if it wouldn't be better to copy the TARGET_FILE (so.X.X.X) to match the name in the addon.xmls. (f.e. `library_linux="audiodecoder.timidity.so.1.0.3"`).